docker/scripts/hardware/utils.py

```diff
@@ -110,9 +110,15 @@ def get_smart_devices():
 
         dev, _, disk_type = line.split(" ")[0:3]
 
-        if "megaraid" not in disk_type:
+        if "megaraid" in disk_type:
             continue
 
+        if "nvme" in disk_type:
+            smart_map[disks[counter] + "n1"] = disk_type
+        else:
+            smart_map[disks[counter]] = disk_type
```

Comment:
I'm still confused by this counter-based business...it seems like we're relying on the relative ordering of what `os.listdir()` returns vs. what `smartctl --scan-open` prints and effectively sort of manually `zip()`-ing them together?  Except I think at least `os.listdir()`'s return wouldn't have any particular ordering guaranteed, so I'm not sure if it's just been working by luck or I'm misunderstanding what's going on...perhaps @splaspood can comment?

Reply:
And further, with this version we don't have the regex tweak, so the `disks` list is still only going to have `sd*` entries, right?  So I'm not sure how the `+ "n1"` part could have done anything useful, since I don't see how `disks[counter]` could produce e.g. `"nvme0"`...

Reply:
I guess perhaps my high-level question (for @splaspood I guess?) is why we're using `os.listdir()` at all instead of just using the device names from the smartctl output...

Reply:
If only I could recall...  I have to assume that at some point, or under certain conditions, smartctl wasn't presenting all devices...   but it's been so long since I wrote this that I no longer recall my reasoning.